flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/utils/TableEnvUtils.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.utils;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.api.internal.TableEnvironmentInternal;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.catalog.CatalogManager;
+import org.apache.flink.table.catalog.ConnectorCatalogTable;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.catalog.UnresolvedIdentifier;
+import org.apache.flink.table.sinks.TableSink;
+import org.apache.flink.table.sources.TableSource;
+
+import java.util.Optional;
+
+/**
+ * Utility methods to register table source/sink as a Table in {@link TableEnvironment}'s catalog.
+ */
+@Internal
+public class TableEnvUtils {
+	/**
+	 * Register a table source as a Table under the given name in this
+	 * {@link TableEnvironment}'s catalog.
+	 */
+	public static void registerTableSource(
+			TableEnvironment tEnv,
+			String name,
+			TableSource<?> tableSource,
+			boolean isBatch) {
+		if (tEnv instanceof TableEnvironmentInternal) {
```

Comment:
check and throw exception first, and do other normal logic without this big brace